### ISSUE TITLE
Returns UNSATISFIABLE for solvable CSP

### ISSUE THREAD

**Issue by AML14** [MiniZinc/libminizinc#359]
Given the following mzn:

```
var 0..10: p1;
var 0..10: p2;
var 0..10: p3;
var 0..10: p4;

constraint (not ((p1==1 /\ p2==0) \/ (p3==1 /\ p4==0)));
constraint p1 = 1;
solve satisfy;
```

It returns:

```
  WARNING: model inconsistency detected
=====UNSATISFIABLE=====
```

Whereas when expressing the problem in a different way, like:

```
var 0..10: p1;
var 0..10: p2;
var 0..10: p3;
var 0..10: p4;

constraint (not (p1==1 /\ p2==0) /\ not (p3==1 /\ p4==0));
constraint p1 = 1;
solve satisfy;
```

It returns a (correct) solution:

```
p1 = 1;
p2 = 10;
p3 = 10;
p4 = 10;
----------
```

Note that the only change I did is transforming a `NOT (X OR Y)` as a `(NOT X) AND (NOT Y)`.

Am I missing something? Just for the record, I am using the latest version of MiniZinc. Interestingly, I tried with the fzn-gecode.exe from version 0.9 and it solves the first mzn!

**Comment by AML14** [comment]
Update: the command I'm running (with the legacy version) to make it work is the following:

```
mzn2fzn.exe constraints.mzn && fzn-gecode.exe constraints.fzn | solns2out.exe constraints.ozn
```

Is there any way to execute an equivalent command to that one but with the newest version?

**Comment by guidotack** [comment]
Yes, you can simply use the `minizinc` command like this:

```
minizinc --solver gecode -c constraints.mzn && fzn-gecode constraints.fzn | minizinc --ozn-file constraints.ozn
```

We've fixed the bug, it should appear on the develop branch soon (and the fix will be in the next release).

**Comment by AML14** [comment]
Thanks for the answer and the fix, I hope to see the next release soon. In the meantime, is there an easy way I can build a minizinc.exe with the latest code available? I'm stuck unless I use a very old version of MiniZinc, which suffers from other issues.

**Comment by cyderize** [comment]
The fix landed in 2.4.2 which has been released.

You should be able to follow the instructions in the documentation at https://www.minizinc.org/doc-2.4.2/en/installation_detailed_windows.html if you wish to compile the latest minizinc yourself.

**Comment by AML14** [comment]
Thanks for letting me know.
